instrumentation/scala-fork-join-2.8/javaagent/src/test/scala/io/opentelemetry/javaagent/instrumentation/scalaexecutors/ScalaInstrumentationTest.scala

```diff
@@ -180,34 +180,39 @@ class ScalaInstrumentationTest {
     testing.waitAndAssertTraces(
       new Consumer[TraceAssert] {
         override def accept(trace: TraceAssert): Unit =
-          trace.hasSpansSatisfyingExactly(
-            new Consumer[SpanDataAssert] {
-              override def accept(span: SpanDataAssert): Unit =
-                span
+          trace.satisfiesExactlyInAnyOrder(
```

Comment:
Do the `Await` above not guarantee the ordering? Was hoping to reason the order out, if it's not possible we can remove the `Await`s above.

Reply:
Maybe `Await.result` completes before `onSuccess` has completed?

Reply:
https://stackoverflow.com/questions/34808024/how-do-i-wait-for-a-scala-futures-onsuccess-callback-to-complete